modules/integration/tests-integration/tests-service/src/test/java/org/wso2/carbon/ml/MLTestUtils.java

```diff
@@ -171,41 +175,111 @@ public static boolean checkModelStatus(String modelName, MLHttpClient mlHttpclie
+     *
+     * @param algorithmName Name of the learning algorithm
+     * @param algorithmType Type of the learning algorithm
+     * @param response Response attribute
+     * @param trainDataFraction Fraction of data from the dataset to be trained with
+     * @param normalLabels normal label values
+     * @param newNormalLabel normal label
+     * @param newAnomalyLabel anomaly label
+     * @param normalization normalization option
+     * @param projectID ID of the project
+     * @param versionSetId Additional information about the name
+     * @throws MLHttpClientException
+     * @throws IOException
+     * @throws JSONException
+     */
+    public static String createModelWithConfigurations(String algorithmName, String algorithmType, String response,
+            String trainDataFraction, String normalLabels, String newNormalLabel, String newAnomalyLabel,
+            String normalization, int projectID, int versionSetId, MLHttpClient mlHttpclient)
+                    throws MLHttpClientException, IOException, JSONException {
+        analysisName = algorithmName + versionSetId;
+
+        // Create an analysis
+        mlHttpclient.createAnalysis(analysisName, projectID);
+        analysisId = mlHttpclient.getAnalysisId(projectID, analysisName);
+        return createModelWithConfigurations(algorithmName, algorithmType, response, trainDataFraction, normalLabels,
+                newNormalLabel, newAnomalyLabel, normalization, projectID, versionSetId, analysisId, mlHttpclient);
+    }
+
+    /**
+     * Create a model with given configurations of the model to be trained
+     *
+     * @param algorithmName Name of the learning algorithm
+     * @param algorithmType Type of the learning algorithm
+     * @param response Response attribute
+     * @param trainDataFraction Fraction of data from the dataset to be trained with
+     * @param projectID ID of the project
+     * @param versionSetId Additional information about the name
+     * @throws MLHttpClientException
+     */
+    public static String createModelWithConfigurations(String algorithmName, String algorithmType, String response,
+            String trainDataFraction, int projectID, int versionSetId, int analysisId, MLHttpClient mlHttpclient)
+                    throws MLHttpClientException, IOException, JSONException {
+        mlHttpclient.setFeatureDefaults(analysisId);
+
+        // Set Model Configurations
+        mlHttpclient.setModelConfiguration(analysisId,
+                setModelConfigurations(algorithmName, algorithmType, response, trainDataFraction));
+
+        // Set default Hyper-parameters
+        mlHttpclient.doHttpPost("/api/analyses/" + analysisId + "/hyperParams/defaults", null);
+
+        // Create a model
+        CloseableHttpResponse httpResponse = mlHttpclient.createModel(analysisId, versionSetId);
+        modelName = mlHttpclient.getModelName(httpResponse);
+        modelId = mlHttpclient.getModelId(modelName);
+
```

Comment:
Thanks for pointing it out. Fixed with https://github.com/wso2/product-ml/pull/313/commits/12ea5260eb9f6401a97a560919fb4074235ebeaa